src/main.rs

```diff
@@ -415,6 +418,43 @@ fn handle_config_errors(config: Result<Config, Vec<String>>) -> Option<Config> {
     }
 }
 
+fn read_configs(config_paths: &Vec<PathBuf>) -> Result<Config, Vec<String>> {
+    let mut config = vector::topology::Config::empty();
+    let mut errors = Vec::new();
+
+    config_paths.iter().for_each(|p| {
+        let file = if let Some(file) = open_config(&p) {
+            file
+        } else {
+            errors.push(format!("Config file not found in path: {:?}.", p));
+            return;
+        };
+
+        trace!(
+            message = "Parsing config.",
+            path = ?p
+        );
+
+        match Config::load(file) {
+            Ok(next_config) => match config.append(next_config) {
```

Comment:
minor nit: This seems like the perfect opportunity for a `if let config.append(..)` or also I think you could avoid this second branch by using `match Config::load(file).and_then(Config::append) { ... }`.